### PR TITLE
add callout to avoid profile.yml confusion for cloud users

### PR DESCRIPTION
**Pull request by mirnawong1**
[Docs project](https://www.notion.so/dbtlabs/Evaluate-References-content-v1-3102e75968cd4c8fb0d92d4e3eb54c25?pvs=4)

[Per slack](https://dbt-labs.slack.com/archives/C02NCQ9483C/p1681495304556999), added a callout for the cloud-supported profile pages to clarify that profiles.yml is only for CLI users. this callout also clarifies that cloud users don't need profiles.yml and can connect to their data platform directly on the about platform page. 

Also made tweaks to the 'about cli' page as well since i was in there. 

mini plan:
this week: address Katherine’s feedback, i’ll leave the setup as it, add the callout for the cloud-supported pages. sync w anders to get more context.
next week: then after sync w anders, open another pr where i move the setups section to docs -> set up dbt -> dbt core -> platform setup or something.

## Checklist

- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding new pages (delete if not applicable):
- [ ] Add page to `website/sidebars.js`
- [ ] Provide a unique filename for the new page

Removing or renaming existing pages (delete if not applicable):
- [ ] Remove page from `website/sidebars.js`
- [ ] Add an entry `_redirects`
- [ ] [Ran link testing](https://github.com/dbt-labs/docs.getdbt.com#running-the-cypress-tests-locally) to update the links that point to the deleted page

### CODE REVIEW

File: website/docs/docs/core/about-the-cli.md
Comment (mirnawong1) [inline website/docs/docs/core/about-the-cli.md:14]: removed this as it's not dbt supported. convo [here](https://dbt-labs.slack.com/archives/C02G5V9JBHV/p1682434162754379?thread_ts=1682429905.097399&cid=C02G5V9JBHV)

File: website/docs/reference/warehouse-setups/postgres-setup.md
Comment (runleonarun) [inline website/docs/reference/warehouse-setups/postgres-setup.md]: @mirnawong1 We can ship and then follow up on this, but I think these callouts are a great use case for reusables! It would keep the wording consistent and if we need to add any other context, we can do it in one place.

File: website/sidebars.js
Comment (runleonarun) [inline website/sidebars.js:146]: nice catch!

File: website/docs/reference/profiles.yml.md
Comment (runleonarun) [inline website/docs/reference/profiles.yml.md:6]: Love this having more info here! SO helpful

**Review by runleonarun — DISMISSED** [review]
It looks great! The only non-blocking comment is to change the notes to reusables, but you can follow-up on that if you want to! Approving so you can decide how to approach that feedback.

File: website/docs/reference/warehouse-setups/postgres-setup.md
Comment (mirnawong1) [reply website/docs/reference/warehouse-setups/postgres-setup.md]: sounds good Leona - i'm going to use the snippet and then merge this since you've approved it already (content-wise) i hope that's ok and gets this up for Katherine and our users.

**Comment by mirnawong1** [comment]
merging this as the content was approved and taking @runleonarun 's suggestion to turn it into snippets to reduce manual work and errors. 

hope it was ok to merge this without 2nd approval @runleonarun !

**Pull request merged**
